### PR TITLE
NTLM Support feature

### PR DESCRIPTION
**Pull request by kpanwar**
NTLM Support feature
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r23177512%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r23177759%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r24627803%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r24627804%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23issuecomment-74171029%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23issuecomment-74171029%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Sign-off%22%2C%20%22created_at%22%3A%20%222015-02-12T22%3A51%3A55Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10979061%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ranyelhousieny%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2009762e3f75cd98cc92222c056d9b7d59f1d7cb7c%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%2099%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r23177512%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20text%20will%20not%20be%20translated%20%5C%22%5C%22Enter%20your%20credentials%5C%22%22%2C%20%22created_at%22%3A%20%222015-01-19T18%3A26%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-02-12T22%3A37%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m%3AL1-149%22%7D%2C%20%22Pull%2009762e3f75cd98cc92222c056d9b7d59f1d7cb7c%20ADALiOS/ADALiOS/ADAuthenticationBroker.m%2034%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244%23discussion_r23177759%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Comment%20is%20related%20to%20the%20TLS.%22%2C%20%22created_at%22%3A%20%222015-01-19T18%3A29%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/466805%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/omercs%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-02-12T22%3A37%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADAuthenticationBroker.m%3AL218-230%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244#issuecomment-74171029'>General Comment</a></b>
- <a href='https://github.com/ranyelhousieny'></a> Sign-off
- [x] <a href='#crh-comment-Pull 09762e3f75cd98cc92222c056d9b7d59f1d7cb7c ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m 99'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244#discussion_r23177512'>File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m:L1-149</a></b>
- <a href='https://github.com/omercs'></a> This text will not be translated ""Enter your credentials"
- [x] <a href='#crh-comment-Pull 09762e3f75cd98cc92222c056d9b7d59f1d7cb7c ADALiOS/ADALiOS/ADAuthenticationBroker.m 34'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244#discussion_r23177759'>File: ADALiOS/ADALiOS/ADAuthenticationBroker.m:L218-230</a></b>
- <a href='https://github.com/omercs'></a> Comment is related to the TLS.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/244?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/244?approve=1'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/244'>Refresh</a>.<a href='#crh-update-none'></a>

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    if ([challenge.protectionSpace.authenticationMethod caseInsensitiveCompare:NSURLAuthenticationMethodNTLM] == NSOrderedSame)
+    {
+        @synchronized(self)
+        {
+            if(_conn){
+                _conn = nil;
+            }
+            // This is the client TLS challenge: use the identity to authenticate:
+            AD_LOG_VERBOSE_F(AD_WPJ_LOG, @"Attempting to handle %@ challenge for host: %@", challenge.protectionSpace.authenticationMethod, challenge.protectionSpace.host);
+            NSAlert *alert = [NSAlert alertWithMessageText: @"Enter your credentials"
```
Comment (omercs) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: This text will not be translated ""Enter your credentials"

File: ADALiOS/ADALiOS/ADAuthenticationBroker.m [hunk 2]
```diff
@@ -219,6 +218,12 @@ - (void)start:(NSURL *)startURL
         }
         if ( parent )
         {
+            _ntlmSession = [ADNTLMHandler startWebViewNTLMHandlerWithError:nil];
+            if (_ntlmSession)
+            {
+                AD_LOG_INFO(@"Authorization UI", @"The device is workplace joined. Client TLS Session started.");
```
Comment (omercs) [inline ADALiOS/ADALiOS/ADAuthenticationBroker.m]: Comment is related to the TLS.

File: ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m (hunk 1 above)
Comment (kpanwar) [inline ADAL-OSX/ADAL-OSX/ADNTLMHandlerOSX.m]: :+1:<a href='#crh-update-none'></a>

File: ADALiOS/ADALiOS/ADAuthenticationBroker.m (hunk 2 above)
Comment (kpanwar) [inline ADALiOS/ADALiOS/ADAuthenticationBroker.m]: :+1:<a href='#crh-update-complete'></a>

**Comment by ranyelhousieny** [comment]
Sign-off

**Pull request merged**